opencollective/opencollective-api
# Internal Transferwise Payout MVP

Related to https://github.com/opencollective/opencollective/issues/2836
Fix https://github.com/opencollective/opencollective/issues/2890

## Testing instructions

1. Run `npm run db:migrate`
1. Create a PayoutMethod for your current testing user with
    ```js
    data: {"accountHolderName":"Leo Kewitz","currency":"EUR","type":"iban","legalType":"PRIVATE","details":{"IBAN":"DE89370400440532013000"}}
    ```
1. Create a ConnectedAccount for Open Source Collective Host with
    ```js
    service: 'transferwise'
    token: '33b5e94d-9815-4ebc-b970-3612b6aec332'
    data: {"type":"business","id":220192}
    ```
1. Submit an expense to any collective hosted by Open Source Collective.
    - Select "Bank Transfer" for payout method.
1. As an admin of that collective, approve the expense.
1. As an admin of Open Source Collective, click on the `Pay with Transferwise` button.
    1. The Expense should display _PROCESSING_ status
    1. The transaction should have been created.
    1. The fee should display in the details of the transaction.
1. Run `npm run script cron/hourly/check-pending-transferwise-transactions.js`
    1. It should log the processing expense and mark it as paid if the transfer is completed.
